contracts/poap/src/contract.rs

```diff
@@ -0,0 +1,443 @@
+
+    // Check if the event is started
+    if !event_info.is_started(&env.block.time) {
+        return Err(ContractError::EventNotStarted {
+            current_time: env.block.time,
+            start_time: event_info.start_time,
+        });
+    }
+
+    // Check if the event is ended
+    if event_info.is_ended(&env.block.time) {
+        return Err(ContractError::EventTerminated {
+            current_time: env.block.time,
+            end_time: event_info.end_time,
+        });
+    }
+
+    // Check if the mint is enabled
+    if !bypass_mint_enable && !config.mint_enabled {
+        return Err(ContractError::MintDisabled {});
+    }
+
+    // Check if who is performing the action is the minter
+    if check_authorized_to_mint && info.sender != config.minter && info.sender != config.admin {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Check per address limit
+    let recipient_mint_count = (MINTER_ADDRESS
+        .key(recipient_addr.clone())
+        .may_load(deps.storage)?)
+    .unwrap_or(0);
+
+    if recipient_mint_count >= config.per_address_limit {
+        return Err(ContractError::MaxPerAddressLimitExceeded {
+            recipient_addr: recipient_addr.to_string(),
+        });
+    }
+
+    // Get the nex poap id
+    let poap_id = NEXT_POAP_ID.may_load(deps.storage)?.unwrap_or(1);
+
+    // Create the cw721 message to send to mint the poap
+    let mint_msg = Cw721ExecuteMsg::Mint(MintMsg::<String> {
+        token_id: poap_id.to_string(),
+        owner: recipient_addr.to_string(),
+        token_uri: Some(event_info.base_poap_uri),
+        extension: event_info.event_uri,
+    });
+
+    let cw721_address = CW721_ADDRESS.load(deps.storage)?;
+    let wasm_execute_mint_msg = wasm_execute(cw721_address, &mint_msg, vec![])?;
+
+    // Update the next poap id state
+    let new_poap_id = poap_id + 1;
+    NEXT_POAP_ID.save(deps.storage, &new_poap_id)?;
+    // Save the new mint count for the sender's address
+    let new_recipient_mint_count = recipient_mint_count + 1;
+    MINTER_ADDRESS.save(
+        deps.storage,
+        recipient_addr.clone(),
+        &new_recipient_mint_count,
+    )?;
+
+    Ok(Response::new()
+        .add_attribute("action", action)
+        .add_attribute("sender", info.sender)
+        .add_attribute("recipient", recipient_addr.to_string())
+        .add_attribute("poap_id", poap_id.to_string())
+        .add_message(wasm_execute_mint_msg))
+}
+
+fn execute_update_event_info(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    start_time: Timestamp,
+    end_time: Timestamp,
+) -> Result<Response, ContractError> {
+    let mut event_info = EVENT_INFO.load(deps.storage)?;
+
+    // Check that is the event creator that is changing the event time frame
+    if event_info.creator != info.sender {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Check that the event is not ended
+    if event_info.is_ended(&env.block.time) {
```

Comment:
It can be removed since `end_time` must be after the `start_time`

Reply:
Yes, that control exists only to return the corect error to the client otherwise the contract would return the `EventStarted` error even when the event is finished.